api/src/main/java/org/openmrs/module/patientflags/api/impl/FlagServiceImpl.java

```diff
@@ -224,7 +240,17 @@ public void purgeFlag(Integer flagId) {
 		// then refresh the cache
 		refreshCache();
 	}
-	
+
+	public void retireFlag(Flag flag, String retiredReason) {
+		if (StringUtils.isNotBlank(retiredReason)) {
+			flag.setRetired(true);
+			flag.setRetireReason(retiredReason);
+			saveFlag(flag);

```

Comment:
The above should be Context.getService(...).saveFlag()
That way you do not by pass AOP

Reply:
@dkayiwa Oh got you - I was wondering why that was a better call than the method in the class :-)